modules/command-engine/core/src/main/scala/surge/internal/core/SurgePartitionRouterImpl.scala

```diff
@@ -27,25 +34,78 @@ private[surge] final class SurgePartitionRouterImpl(
     system: ActorSystem,
     partitionTracker: KafkaConsumerPartitionAssignmentTracker,
     businessLogic: SurgeModel[_, _, _, _],
+    aggregateKafkaStreamsImpl: AggregateStateStoreKafkaStreams[JsValue],
     regionCreator: PersistentActorRegionCreator[String],
-    signalBus: HealthSignalBusTrait)
+    signalBus: HealthSignalBusTrait,
+    isAkkaClusterEnabled: Boolean)
     extends SurgePartitionRouter
     with HealthyComponent
     with Controllable {
   implicit val executionContext: ExecutionContext = system.dispatcher
   private val log = LoggerFactory.getLogger(getClass)
 
-  private val shardRouterProps = KafkaPartitionShardRouterActor.props(
-    config,
-    partitionTracker,
-    businessLogic.partitioner,
-    businessLogic.kafka.stateTopic,
-    regionCreator,
-    RoutableMessage.extractEntityId)(businessLogic.tracer)
+  override val actorRegion: ActorRef = {
+    if (isAkkaClusterEnabled) {
+      createAkkaCluster()
+    } else {
+      createShardRouter()
+    }
+  }
+
+  log.info(s"Shard region: $actorRegion")
+
+  private def createShardRouter(): ActorRef = {
+    val shardRouterProps = KafkaPartitionShardRouterActor.props(
+      config,
+      partitionTracker,
+      businessLogic.partitioner,
+      businessLogic.kafka.stateTopic,
+      regionCreator,
+      RoutableMessage.extractEntityId)(businessLogic.tracer)
 
-  private val routerActorName = s"${businessLogic.aggregateName}RouterActor"
-  private val shardRouter = system.actorOf(shardRouterProps, name = routerActorName)
-  override val actorRegion: ActorRef = shardRouter
+    val routerActorName = s"${businessLogic.aggregateName}RouterActor"
+    val shardRouter = system.actorOf(shardRouterProps, name = routerActorName)
+    shardRouter
+  }
+
+  private def createAkkaCluster(): ActorRef = {
+    val groupId = businessLogic.kafka.streamsApplicationId
+    val consumerSettings = ConsumerSettings(system, new StringDeserializer, new ByteArrayDeserializer)
+      .withBootstrapServers(config.getString("kafka.brokers"))
+      .withGroupId(groupId)
+      .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+      .withStopTimeout(0.seconds)
```

Comment:
This is one of the interesting pieces I'm curious on. The Alpakka Kafka docs have the copartitioning based on the input topic so you can leverage the same consumer settings as your consumer, but with Surge we're copartitioning based on the Surge Kafka Streams consumer (totally outside of Akka). I'm wondering if under the hood the Akka KafkaClusterSharding is doing something special inside the consumer group we should be aware of or if it just leverages the groupId (and any connection related settings) to grab what it needs?

Reply:
Also I think we'll need to plumb authentication related settings to this to support auth for Kafka

Reply:
Yeah, it just leverages connection-related settings. And basically, we use this kafka consumer setting to extract the number of partitions for the state topic.